src/modules/hype_chat/index.js

```diff
@@ -1,26 +1,67 @@
-import {PlatformTypes} from '../../constants.js';
+import {PlatformTypes, SettingIds} from '../../constants.js';
 import dom from '../../observers/dom.js';
+import settings from '../../settings.js';
 import {loadModuleForPlatforms} from '../../utils/modules.js';
 import twitch from '../../utils/twitch.js';
 import chat from '../chat/index.js';
+import styles from './styles.module.css';
 
 const HYPE_CHAT_WRAPPER_SELECTOR = '.paid-pinned-chat-message-content-wrapper';
 const HYPE_CHAT_MESSAGE_SELECTOR = 'span[data-a-target="chat-message-text"]';
 
+const CHAT_TEXT_AREA_SELECTOR = '.chat-input__textarea';
+const STREAM_CHAT_SELECTOR = '.stream-chat';
+
+let removeHypeChatListener = null;
 class HypeChatModule {
   constructor() {
-    dom.on(HYPE_CHAT_WRAPPER_SELECTOR, (node, isConnected) => {
+    this.load();
+    settings.on(`changed.${SettingIds.HYPE_CHAT}`, () => this.load());
+    dom.on(CHAT_TEXT_AREA_SELECTOR, (_, isConnected) => {
```

Comment:
let's refactor this a bit.. dom watcher doesn't need to be loaded all the time, only if the setting is disabled. it would be great to pull this out into a method like we do for the other watcher too.